src/batou/secrets/__init__.py

```diff
@@ -66,8 +73,8 @@ def add_secrets_to_environment(environment):
                         environment.secret_data.update(v.value.split())
 
         # additional_secrets
-        for path in iter_other_secrets(environment.name):
-            secret_name = path.name.replace("secret-", "", 1)
+        for path in iter_other_secrets(environment.name, secrets_type):
+            secret_name = secret_name_from_path(path)
```

Comment:
Maybe all those helper methods could be collected into a coherent object that is related to an environment. This likely requires an inside-out refactoring of the current glue code and we should end up with an API based on `environment.secrets` that hides implementation-specific details (age vs. gpg, and also whether those are individual files or maybe some other mechanism that could be using not even files (but maybe directly store stuff in 1p?) or single large file instead of individual files).

Having that as a proper API should also make migrations between one store and another very simple (like: read from implementation a, write to implementation b, ask implementation a to clean up at the right time)

So, in the end both the secrets editor and the add_secrets_to_environment should be using that API.

Functions I see that could be hidden behind or be part of the API:

* iter_other_secrets
* secret_name_from_path (path names like should be completely hidden in the implementation)
* get_secret_config_from_environment_name
* get_secrets_type

Even the editor should likely be hidden as an implementation detail. (Think: using 1p as the implementation and not even having files to edit)